### ISSUE TITLE
Is it possible to input a geometric proxy to help guide meshing?

### ISSUE THREAD

**Issue by DanAndersen** [alicevision/meshroom#402]
Hi,
I have a somewhat unusual question. I have acquired an input scene with ~150 photos, which have been well matched and aligned. 
![issue_1](https://user-images.githubusercontent.com/2023351/53106948-129d0d80-3502-11e9-8e50-6f79aca3465d.PNG)

Due to some thin and feature-poor regions of the scene, the meshing stage generates poor results (see the underside and edges of the table, leading to distorted geometry under the floor).

![issue_2](https://user-images.githubusercontent.com/2023351/53106949-129d0d80-3502-11e9-9619-ca89c6724074.PNG)

However, I have a rough geometric model of the scene acquired using active methods (the depth sensors on an AR HMD).

![issue_3](https://user-images.githubusercontent.com/2023351/53106950-129d0d80-3502-11e9-8a7a-cb4a2a17f74f.PNG)

I am wondering if there is any feature in meshroom or similar software that would allow me to inject the rough geometric model as a constraint, to encourage the meshing to generate a refined model that is more physically plausible given this geometric proxy.

**Comment by fabiencastan** [comment]
There is no such feature yet.

One thing that would be nice to add in Meshroom is the ability to use RGBD images in input and use the provided depth maps in the dense part of the pipeline.
There was a beginning of work in this direction to update the scene scale (to have depth values in the same coordinate system) to be able to use the provided depth maps, but that's far from being usable:
https://github.com/alicevision/AliceVision/pull/489

Best,

**Comment by DanAndersen** [comment]
Thank you for getting back to me so quickly; is the potential future feature (to add RGBD images as input) connected with what is described in issue #399?

**Comment by fabiencastan** [comment]
Yes, it's related. The only difference is that if the depth maps are rendered (like in Issue 399) the scale of the scene will the same between the rendered depth maps and the reconstructed cameras.
If the depth maps comes from RGBD sensors and the reconstructed cameras are in a random coordinate system, we need to adjust the scale between them. That should be the main challenge.

**Comment by hargrovecompany** [comment]
I think you are way short of the number of photos that you really need to recreate that environment.

**Comment by Sudoki** [comment]
is this feature still being considered?

**Comment by fabiencastan** [comment]
AFAIK, there is no active development on this topic, but it is an interesting subject and we would be happy to support any initiative in this area.

**Comment by kenchikuliu** [comment]
> 
> 
> Hi,
> I have a somewhat unusual question. I have acquired an input scene with ~150 photos, which have been well matched and aligned.
> ![issue_1](https://user-images.githubusercontent.com/2023351/53106948-129d0d80-3502-11e9-8e50-6f79aca3465d.PNG)
> 
> Due to some thin and feature-poor regions of the scene, the meshing stage generates poor results (see the underside and edges of the table, leading to distorted geometry under the floor).
> 
> ![issue_2](https://user-images.githubusercontent.com/2023351/53106949-129d0d80-3502-11e9-9619-ca89c6724074.PNG)
> 
> However, I have a rough geometric model of the scene acquired using active methods (the depth sensors on an AR HMD).
> 
> ![issue_3](https://user-images.githubusercontent.com/2023351/53106950-129d0d80-3502-11e9-8a7a-cb4a2a17f74f.PNG)
> 
> I am wondering if there is any feature in meshroom or similar software that would allow me to inject the rough geometric model as a constraint, to encourage the meshing to generate a refined model that is more physically plausible given this geometric proxy.

Hi, great topic. I also consider this, how about now? Is there any feature in meshroom or similar software that would allow me to inject the rough geometric model as a constraint, to encourage the meshing to generate a refined model that is more physically plausible given this geometric proxy?

**Comment by natowi** [comment]
> Is there any feature in meshroom or similar software

No and not that I know of.
